drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableModelIndexVisitor.java

```diff
@@ -54,83 +54,95 @@
-            visit( (BRLConditionColumn) o );
-        } else if ( o instanceof ConditionCol52 ) {
-            visit( (ConditionCol52) o );
-        } else if ( o instanceof BRLActionColumn ) {
-            visit( (BRLActionColumn) o );
-        } else if ( o instanceof ActionInsertFactCol52 ) {
-            visit( (ActionInsertFactCol52) o );
-        } else if ( o instanceof ActionSetFieldCol52 ) {
-            visit( (ActionSetFieldCol52) o );
-        } else if( o instanceof Imports ) {
-           visit( (Imports) o );
+    private void visit(final Object o) {
+        if (o instanceof GuidedDecisionTable52) {
+            visit((GuidedDecisionTable52) o);
+        } else if (o instanceof AttributeCol52) {
+            visit((AttributeCol52) o);
+        } else if (o instanceof Pattern52) {
+            visit((Pattern52) o);
+        } else if (o instanceof BRLConditionColumn) {
+            visit((BRLConditionColumn) o);
+        } else if (o instanceof ConditionCol52) {
+            visit((ConditionCol52) o);
+        } else if (o instanceof BRLActionColumn) {
+            visit((BRLActionColumn) o);
+        } else if (o instanceof ActionInsertFactCol52) {
+            visit((ActionInsertFactCol52) o);
+        } else if (o instanceof ActionSetFieldCol52) {
+            visit((ActionSetFieldCol52) o);
+        } else if (o instanceof Imports) {
+            visit((Imports) o);
         }
     }
 
-    private void visit( final GuidedDecisionTable52 o ) {
+    private void visit(final GuidedDecisionTable52 o) {
         //Add Imports
-        visit( o.getImports() );
+        visit(o.getImports());
         //Add attributes
-        for ( AttributeCol52 c : o.getAttributeCols() ) {
-            visit( c );
+        for (AttributeCol52 c : o.getAttributeCols()) {
+            visit(c);
         }
         //Add Types and Fields used by Conditions
-        for ( CompositeColumn<? extends BaseColumn> c : o.getConditions() ) {
-            visit( c );
+        for (CompositeColumn<? extends BaseColumn> c : o.getConditions()) {
+            visit(c);
         }
         //Add Types and Fields used by Actions
-        for ( ActionCol52 c : o.getActionCols() ) {
-            visit( c );
+        for (ActionCol52 c : o.getActionCols()) {
+            visit(c);
         }
         //Add rule names
+        for (List<DTCellValue52> row : model.getData()) {
+            final String ruleName = getPackagePrefix() + "Row " + row.get(0).getNumericValue().longValue() + " " + model.getTableName();
+            addResource(ruleName,
+                        ResourceType.RULE);
+        }
+        //Add parent rule name
         final String parentRuleName = model.getParentName();
-        for ( List<DTCellValue52> row : model.getData() ) {
-            final String ruleName = "Row " + row.get( 0 ).getNumericValue().longValue() + " " + model.getTableName();
-            addResourceReference(ruleName, ResourceType.RULE);
```

Comment:
This was the problem. We were adding a "reference to a rule" not a "rule" itself.